File: 50 Coding Challenge Part I/TypeScript/Challenge13.ts
// Challenge 13: Find the maximum number in an array of numbers

function maxNum (arr: number[]): number {
    let max: number = arr[0];

    for (let ar of arr){
        if (ar > max){
            max = ar;
        }
    }

    return max;
}

// Driver Method
let arr2 = [2, 3, -1, 5, 7, 9, 10, 15, 95]
console.log(`Maximum Number: ${maxNum(arr2)}`)

File: 50 Coding Challenge Part I/dart/lib/challenge13.dart
/*  
  Coding Challenge 13: Find the maximum number in an array of numbers

  @author: mofirojean
*/

int arrayMaxA(List<int> arr) {
  int max = arr[0];
  for (var i = 1; i < arr.length; i++) {
    if (arr[i] > max) max = arr[i];
  }
  return max;
}

int arrayMaxB(List<int> arr) {
  int max = 0;
  for (var num in arr) {
    if (num > max) max = num;
  }
  return max;
}

